### PR TITLE
[Customer Portal][FE][web] Refactor: related case naming + code block parsing improvements + UI enhancements

### PR DESCRIPTION
**Pull request by dileepapeiris**
### Description

This pull request introduces several improvements and refactors across the support case details and related features, focusing on consistent naming, improved code block handling in messages, and better display of preferred call times. The most significant changes are grouped below:

**Refactor and Consistency in Related Case Naming:**
- Renamed all usages of `parentCase`/`parentCaseId` to `relatedCase`/`relatedCaseId` in types, navigation state, and API requests for clarity and consistency. This affects types like `CaseDetails`, `CreateCaseRequest`, `RelatedCaseState`, and related components and tests. [[1]](diffhunk://#diff-624c276530bd2d9bcc739ea2890edeb9311826df13dabd7a17a020bc14b50077L218-R218) [[2]](diffhunk://#diff-624c276530bd2d9bcc739ea2890edeb9311826df13dabd7a17a020bc14b50077L365-R365) [[3]](diffhunk://#diff-510a542f9d15f7261c63841c40d3f3c2b27d84c773785c1ec4d4f7fffefb51a2L41-R41) [[4]](diffhunk://#diff-369e3edc78ac64842b24a30d6733cb8cca7d4b288418768e0880fbc70ddbca8fL42-R42) [[5]](diffhunk://#diff-f894823d28bd73fbf8848b5cee3c5cf26d01c94c28f077fcedb6f7e6877f3fb8L47-R47) [[6]](diffhunk://#diff-654fdd7d127ef65b62504eaa3957e6f4fbaf9083d2931bd1453bbd186db1099eL25-R25) [[7]](diffhunk://#diff-0ab1afa5ae279779dbf505e18aa51ff92041021bf0c5993b902c6be4772f4385L118-R118) [[8]](diffhunk://#diff-06948352d8a7ba2500b59f7033bc222c92d269eca28ce90788988a8c207ba13cL93-R93) [[9]](diffhunk://#diff-77ba7db2416ae0479c6bd3f70c06ba28b4a9b6ac99f16572943de1a4cf6dbd68L778-R779)

**Enhanced Code Block Handling in Support Messages:**
- Improved the `convertCodeTagsToHtml` and `stripAllCodeBlocks` utility functions to normalize and correctly handle escaped or orphaned `[code]` tags, ensuring cleaner rendering and text extraction. [[1]](diffhunk://#diff-aaee3b216d9e9e36d662afa6ae27f05490b4f69615370e10c06b3e695cb637dcL1133-R1144) [[2]](diffhunk://#diff-aaee3b216d9e9e36d662afa6ae27f05490b4f69615370e10c06b3e695cb637dcL1152-R1164)
- Added comprehensive tests for edge cases in code block parsing, such as handling escaped markers and orphan tags. [[1]](diffhunk://#diff-082486c85daf57ee9724dd5154eab4df31208098f5687099b08a8151e3caa75eR713-R724) [[2]](diffhunk://#diff-082486c85daf57ee9724dd5154eab4df31208098f5687099b08a8151e3caa75eL721-R749)

**UI Improvements for Message and Call Time Display:**
- Updated the chat message card styling to better handle code blocks, ensuring proper width, background color, and code formatting.
- Refactored the display of preferred call times in `CallRequestCard` to present them as a list when multiple times are provided and improved the formatting logic for clarity. [[1]](diffhunk://#diff-3538e1b3c5433b323e4c57a48e7d973cd48975fd97216d59ff75351d0f77d142L48-R53) [[2]](diffhunk://#diff-3538e1b3c5433b323e4c57a48e7d973cd48975fd97216d59ff75351d0f77d142R89) [[3]](diffhunk://#diff-3538e1b3c5433b323e4c57a48e7d973cd48975fd97216d59ff75351d0f77d142L263-R280)

These changes collectively improve code maintainability, user experience, and the robustness of message parsing and display.

## Release Notes

* **Bug Fixes**
  * Improved code block handling to properly normalize escaped markers and edge cases for better content display

* **Style**
  * Enhanced styling for code and pre-formatted text blocks with improved content sizing and background colors

* **UI Improvements**
  * Preferred times now display as a formatted bullet list for improved readability

### CODE REVIEW

**Review by cloby99 — APPROVED** [review]

File: apps/customer-portal/webapp/src/features/support/components/case-details/calls-tab/CallRequestCard.tsx [hunk 1]
```diff
@@ -260,9 +261,23 @@ export default function CallRequestCard({
...
+                sx={{
+                  m: 0,
+                  pl: 2.5,
+                }}
+              >
+                {preferredTimeLines.map((timeValue) => (
+                  <Box component="li" key={timeValue}>
+                    <Typography variant="body2">{timeValue}</Typography>
+                  </Box>
+                ))}
```
[the remark this answers is not in the document]
Comment (dileepapeiris) [reply apps/customer-portal/webapp/src/features/support/components/case-details/calls-tab/CallRequestCard.tsx:278]: Will fix in next PR

[the remark this answers is not in the document]
Comment (dileepapeiris) [reply apps/customer-portal/webapp/src/features/support/components/case-details/calls-tab/CallRequestCard.tsx:278]: Fixed via  https://github.com/wso2-open-operations/cs-tools/pull/505
